libopae/plugins/xfpga/sysfs.c

```diff
@@ -1874,6 +1918,54 @@ fpga_result make_sysfs_group(char *sysfspath, const char *name,
 	return res;
 }
 
+
+fpga_result make_sysfs_array(char *sysfspath, const char *name,
+			     fpga_object *object, int flags, fpga_handle handle,
+			     max_path_t objects[], size_t num_objects)
+{
+	fpga_result res = FPGA_OK;
+	size_t i = 0;
+	struct _fpga_object *array = alloc_fpga_object(sysfspath, name);
+	char *oname = NULL;
+	if (!array) {
+		FPGA_ERR(
+			"Error allocating memory for container of fpga_objects");
+		return FPGA_NO_MEMORY;
+	}
+	array->objects = calloc(num_objects, sizeof(fpga_object));
+	if (!array->objects) {
+		FPGA_ERR("Error allocating memory for array of fpga_objects");
+		return FPGA_NO_MEMORY;
+	}
+
+	array->handle = handle;
+	array->type = FPGA_SYSFS_LIST;
+	array->size = num_objects;
+	for (i = 0; i < num_objects; ++i) {
+		oname = strrchr(objects[i], '/');
+		if (!oname) {
+			FPGA_ERR("Error with sysfs path: %s", objects[i]);
+			res = FPGA_EXCEPTION;
+			goto out_err;
+		}
+		res = make_sysfs_object(objects[i], oname+1, &array->objects[i],
+					flags & ~FPGA_OBJECT_GLOB, handle);
+		if (res) {
+			goto out_err;
+		}
+	}
+	*object = (fpga_object)array;
+	return res;
+out_err:
+	if (array->objects) {
+		free(array->objects);
```

Comment:
Is there any memory allocation in each array->objects[i] that needs to be freed?